paper-api/src/main/java/org/bukkit/entity/HumanEntity.java

```diff
@@ -732,121 +734,56 @@ default void openSign(@NotNull org.bukkit.block.Sign sign) {
      */
     @Nullable
     public default Item dropItem(int slot, int amount) {
-        return this.dropItem(slot, amount, null);
+        return this.dropItem(slot, amount, false, null);
     }
 
     /**
-     * Makes the player drop an item from their inventory based on the equipment slot.
-     *
-     * @param slot            The equipment slot to drop
-     * @param amount          The amount of items to drop from this equipment slot. Values below one always return null
-     * @param entityOperation The function to be run before adding the entity into the world
-     * @return The dropped item entity, or null if the action was unsuccessful
-     */
-    @Nullable
-    public Item dropItem(@NotNull EquipmentSlot slot, int amount, @Nullable Consumer<Item> entityOperation);
-
-    /**
-     * Makes the player drop an item from their inventory based on the equipment slot.
+     * Makes the player drop all items from their inventory based on the inventory slot.
      *
-     * @param slot   The equipment slot to drop
-     * @param amount The amount of items to drop from this equipment slot. Values below one always return null
+     * @param slot The equipment slot to drop
      * @return The dropped item entity, or null if the action was unsuccessful
      */
     @Nullable
-    public default Item dropItem(@NotNull EquipmentSlot slot, int amount) {
-        return this.dropItem(slot, amount, null);
+    public default Item dropItem(int slot) {
+        return this.dropItem(slot, Integer.MAX_VALUE, false, null);
```

Comment:
int max value feels a bit weird
this can probably just be moved to the impl and use the size of the stack

Reply:
Its fine :+1: